### ISSUE TITLE
Error importing device from svg (Trac #137)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#141]
From the error log:

[WARNING]:     The following paths could not be parsed properly have been ignore
d:
    Error parsing problem device.svg:5385, loop needs 3 or more verts
    <path xmlns="http://www.w3.org/2000/svg" xmlns:inkscape="http://www.inkscape
.org/namespaces/inkscape" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="
http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-synt
ax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/19
99/xlink" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" in
kscape:connector-curvature="0" style="fill:none;stroke:http://microfluidics.utoronto.ca/microdrop/ticket/0;stroke-width:0.1;
stroke-miterlimit:4;stroke-dasharray:none" d="M185.46948 288.025L185.46948 288.0
25" id="path5113"/>

Note also that the program shows a warning box, but is left in a bad state with the file dialog still open.

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/137
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "From the error log:\n\n[WARNING]:     The following paths could not be parsed properly have been ignore\nd:\n    Error parsing problem device.svg:5385, loop needs 3 or more verts\n    <path xmlns=\"http://www.w3.org/2000/svg\" xmlns:inkscape=\"http://www.inkscape\n.org/namespaces/inkscape\" xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"\nhttp://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-synt\nax-ns#\" xmlns:svg=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/19\n99/xlink\" xmlns:sodipodi=\"http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd\" in\nkscape:connector-curvature=\"0\" style=\"fill:none;stroke:#000000;stroke-width:0.1;\nstroke-miterlimit:4;stroke-dasharray:none\" d=\"M185.46948 288.025L185.46948 288.0\n25\" id=\"path5113\"/>\n\nNote also that the program shows a warning box, but is left in a bad state with the file dialog still open.", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Error importing device from svg", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2013-02-27T19:16:09", 
    "milestone": "Microdrop 1.0", 
    "owner": "", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20130227T19:16:30: ryan changed attachment from "" to "problem device.svg"

**Comment by cfobel** [comment]
Trac update at 20130227T19:17:55: ryan changed description from:

> From the error log:
> 
> [WARNING]:     The following paths could not be parsed properly have been ignore
> d:
>     Error parsing problem device.svg:5385, loop needs 3 or more verts
>     <path xmlns="http://www.w3.org/2000/svg" xmlns:inkscape="http://www.inkscape
> .org/namespaces/inkscape" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="
> http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-synt
> ax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/19
> 99/xlink" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" in
> kscape:connector-curvature="0" style="fill:none;stroke:#000000;stroke-width:0.1;
> stroke-miterlimit:4;stroke-dasharray:none" d="M185.46948 288.025L185.46948 288.0
> 25" id="path5113"/>

to:

> From the error log:
> 
> [WARNING]:     The following paths could not be parsed properly have been ignore
> d:
>     Error parsing problem device.svg:5385, loop needs 3 or more verts
>     <path xmlns="http://www.w3.org/2000/svg" xmlns:inkscape="http://www.inkscape
> .org/namespaces/inkscape" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="
> http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-synt
> ax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/19
> 99/xlink" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" in
> kscape:connector-curvature="0" style="fill:none;stroke:#000000;stroke-width:0.1;
> stroke-miterlimit:4;stroke-dasharray:none" d="M185.46948 288.025L185.46948 288.0
> 25" id="path5113"/>
> 
> Note also that the program shows a warning box, but is left in a bad state with the file dialog still open.

**Comment by cfobel** [comment]
Trac update at 20130411T17:42:37: 
* ryan commented:

> This was fixed in commit 2e0b0f533289e0ba013581d1389b21cb764cdb07.
> 
> Note that the attached device has still has one path that cannot be properly tessellated.


* ryan changed resolution from "" to "fixed"
* ryan changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "" to "microdrop"
